lib/linux_admin/yum.rb

```diff
@@ -43,11 +43,11 @@ def self.updates_available?(*packages)
       cmd    = "yum check-update"
       params = {nil => packages} unless packages.blank?
 
-      exitstatus = Common.run(cmd, :params => params).exit_status
-      case exitstatus
+      spawn = Common.run(cmd, :params => params)
+      case spawn.exit_status
       when 0;   false
       when 100; true
-      else raise "Error: Exit Code #{exitstatus}"
+      else raise "Error: #{cmd} returns '#{spawn.exit_status}', '#{spawn.error}'"
```

Comment:
There is always something on stdout. Like the list of enabled plug-ins.

In my previous life (around 2010) I have been writing automated tests for yum. At that time the tool was pretty rigid on putting errors on stderr and user messages to stdout.